test/smoke/src/debugIos.test.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import { SpectronApplication } from "./spectron/application";
+import * as assert from "assert";
+import { AppiumHelper, Platform } from "./helpers/appiumHelper";
+import { SmokeTestsConstants } from "./helpers/smokeTestsConstants";
+import { RNworkspacePath } from "./main";
+import { IosSimulatorHelper } from "./helpers/iosSimulatorHelper";
+import { Client, RawResult } from "webdriverio";
+import { sleep, runInParallel } from "./helpers/utilities";
+import { SetupEnvironmentHelper } from "./helpers/setupEnvironmentHelper";
+
+const RN_APP_BUNDLE_ID = "org.reactjs.native.example.latestRNApp";
+const RNDebugConfigName = "Debug iOS";
+// const ExpoDebugConfigName = "Debug in Exponent";
+// Time for OS Debug Test before it reaches timeout
+const debugIosTestTime = SmokeTestsConstants.iosAppBuildAndInstallTimeout + 100 * 1000;
+// Time for iOS Expo Debug Test before it reaches timeout
+// const debugExpoTestTime = smokeTestsConstants.expoAppBuildAndInstallTimeout + 400 * 1000;
+export function setup() {
+    describe("Debugging iOS", () => {
+        let clientInited: Client<RawResult<null>> & RawResult<null>;
+        before(async function () {
+            const app = this.app as SpectronApplication;
+            app.suiteName = "Debugging iOS";
+        });
+
+        after(() => {
+            if (clientInited) {
+                clientInited.closeApp();
+                clientInited.endAll();
+            }
+        });
+
+        it("RN app Debug test", async function () {
+            this.timeout(debugIosTestTime);
+            const app = this.app as SpectronApplication;
+            await app.restart({workspaceOrFolder: RNworkspacePath});
+            await app.workbench.explorer.openExplorerView();
+            await app.workbench.explorer.openFile("App.js");
+            await app.runCommand("cursorTop");
+            console.log("iOS Debug test: App.js file is opened");
+            await app.workbench.debug.setBreakpointOnLine(23);
+            console.log("iOS Debug test: Breakpoint is set on line 23");
+            await app.workbench.debug.openDebugViewlet();
+            await app.workbench.debug.chooseDebugConfiguration(RNDebugConfigName);
+            console.log(`iOS Debug test: Chosen debug configuration: ${RNDebugConfigName}`);
+            // We need to implicitly add target to "Debug iOS" configuration to avoid running additional simulator
+            SetupEnvironmentHelper.addIosTargetToLaunchJson(RNworkspacePath);
+            const waitUntilIosAppIsInstalled = AppiumHelper.waitUntilIosAppIsInstalled(RN_APP_BUNDLE_ID, SmokeTestsConstants.iosAppBuildAndInstallTimeout, 40 * 1000);
+            const startDebugging = async () => {
+                console.log("iOS Debug test: Starting debugging");
+                await app.workbench.debug.startDebugging();
```

Comment:
Are you sure that we need to run this operation in parallel? We can start debugging and then wait when app will be installed. StartDebugging is just clicks on run debug button, so it is a fast operation and we can run them sequentially with waitUntilIosAppIsInstalled.